### ISSUE TITLE
RELAYNETS not honored when login is different from sender

### ISSUE THREAD

**Issue by Sharsie** [Mailu/Mailu#369]
I'm having an issue migrating from previous stable (1.4) to 1.5.1.
My setup uses RELAYNETS to allow one of the services from several of my servers to send out emails using a single account for login. However the service sends out email for different users.
This has worked fine with 1.4 where SMTP connections went directly to the `smtp` container and does not work anymore with 1.5.1 where connections first go through the `front` container.
It might be an issue with my configuration, but I'm not really sure where to go from here or how to debug it further.

In an example, I have the following configuration:

I will use my server's addresses as 1.1.1.1, 2.2.2.2 ...

There is an account user@example.com which can send out emails
Then there are several aliases mapped to external addresses such as
client@example.com -> client@gmail.com
client2@example.com -> user@clients.domain
and so on...
It is important that the emails go out as client@example.com and not user@example.com and it is also important that emails sent to client@example.com are relayed to clients private emails

RELAYNETS=172.16.0.0/12 1.1.1.1 2.2.2.2

The service authenticates as: user@example.com
And sends the email as: client@example.com

```
postfix/smtpd[112]: connect from mailu_front_1.mailu_default[172.19.0.4]
postfix/smtpd[112]: NOQUEUE: reject: RCPT from unknown[1.1.1.1]: 553 5.7.1 <client@example.com>: Sender address rejected: not owned by user user@example.com; from=<client@example.com> to=<out@example.com> proto=ESMTP helo=<subdomain.example.com>
postfix/smtpd[112]: disconnect from unknown[1.1.1.1] ehlo=1 mail=1 rcpt=0/1 quit=1 commands=3/4
```

I have checked inside the smtp container for `mynetworks` and the networks are indeed present

```
/etc/postfix # grep -nr mynetworks .
./main.cf:17:mynetworks = 127.0.0.1/32 [::1]/128 172.16.0.0/12 1.1.1.1 2.2.2.2
```

It ofcourse also works if I disable the login mismatch restrictions
```
/etc/postfix # sed -i 's#reject_sender_login_mismatch,##' master.cf
/etc/postfix # postfix reload
```
```
postfix/smtpd[162]: connect from mailu_front_1.mailu_default[172.19.0.4]
postfix/smtpd[162]: 1FFBE87C1D: client=unknown[1.1.1.1], sasl_method=XCLIENT, sasl_username=user@example.com
postfix/cleanup[165]: 1FFBE87C1D: replace: header Received: from subdomain.example.com (unknown [1.1.1.1])??by mail.example.com (Postfix) with ESMTPA id 1FFBE87C1D??for <out@example.com>; Sat, 13 Jan 2018 20:14:54 +0000 (UTC) from unknown[1.1.1.1]; from=<client@example.com> to=<out@example.com> proto=ESMTP helo=<subdomain.example.com>: Received: from authenticated-user (PRIMARY_HOSTNAME [PUBLIC_IP])??by mail.example.com (Postfix) with ESMTPA id 1FFBE87C1D??for <out@example.com>; Sat, 13 Jan 2018 20:14:54 +0000 (UTC)
postfix/cleanup[165]: 1FFBE87C1D: message-id=<7ca2ecf3cdbb3e6ac708744cc10321a6@subdomain.example.com>
postfix/qmgr[161]: 1FFBE87C1D: from=<client@example.com>, size=3781, nrcpt=1 (queue active)
postfix/smtpd[162]: disconnect from unknown[1.1.1.1] ehlo=1 mail=1 rcpt=1 data=1 quit=1 commands=5
postfix/lmtp[166]: 1FFBE87C1D: to=<other@example.com>, orig_to=<out@example.com>, relay=imap[172.19.0.8]:2525, delay=0.22, delays=0.15/0/0.01/0.05, dsn=2.0.0, status=sent (250 2.0.0 <other@example.com> sHVLEb5oWlozAAAAMdehtw Saved)
postfix/qmgr[161]: 1FFBE87C1D: removed
```

Is there anything I can do to help debug this or has something changed in 1.5.1 which forbids this setup?

**Comment by kaiyou** [comment]
Indeed the behavior changed since version 1.5: authenticated users must have a matching email address, no matter what. Therefore, for ``user@example.com`` to be able to impersonate ``client@example.com``, then ``client@example.com`` should include ``user@example.com`` in the alias.

On the other hand, ``RELAYNETS`` lists the networks that can send *any* email *without* authentication. You should be able to impersonate any email address from these IPs as long as you do not authenticate and use TCP port 25.

**Comment by Sharsie** [comment]
Thanks @kaiyou for explaining the behaviour, sorry for being silent for so long, it's been crazy busy :\

So I tried logging in without authentication on 25, but then got refused by postgrey, but I cannot find any configuration in mailu which would allow me to whitelist IPs in greylist (ehm:)

Is this something that would need to be handled on your side or can I do something about it?

postfix:
```
postfix/smtpd[2517]: connect from mailu_front_1.mailu_default[172.19.0.3]
postfix/smtpd[2517]: 0E14041643: client=unknown[1.1.1.1]
postfix/cleanup[2520]: 0E14041643: message-id=<72e3cbe057d72971f453d93d31206447@subdomain.example.com>
postfix/cleanup[2520]: 0E14041643: milter-reject: END-OF-MESSAGE from unknown[1.1.1.1]: 4.7.1 Try again later; from=<client@example.com> to=<out@example.com> proto=ESMTP helo=<subdomain.example.com>
postfix/smtpd[2517]: disconnect from unknown[1.1.1.1] ehlo=1 mail=1 rcpt=1 data=0/1 quit=1 commands=4/5
```
rspamd:
```
2018-01-30 15:14:11 #6(rspamd_proxy) <6b155c>; proxy; proxy_accept_socket: accepted milter connection from 172.19.0.9 port 34486
2018-01-30 15:14:11 #6(rspamd_proxy) <6b155c>; milter; rspamd_milter_process_command: got connection from 172.19.0.3:38866
2018-01-30 15:14:11 #6(rspamd_proxy) <6b155c>; proxy; proxy_milter_finish_handler: finished milter connection
2018-01-30 15:14:11 #6(rspamd_proxy) <cfd4cb>; proxy; proxy_accept_socket: accepted milter connection from 172.19.0.9 port 34488
2018-01-30 15:14:11 #6(rspamd_proxy) <cfd4cb>; milter; rspamd_milter_process_command: got connection from 1.1.1.1:38866
2018-01-30 15:14:11 #6(rspamd_proxy) <cfd4cb>; proxy; rspamd_message_parse: loaded message; id: <72e3cbe057d72971f453d93d31206447@subdomain.example.com>; queue-id: <0E14041643>; size: 1157; checksum: <2b01e01cfe056645f51157bda188d6f3>
2018-01-30 15:14:11 #6(rspamd_proxy) <cfd4cb>; surbl; surbl_test_url: disable surbl multi.uribl.com as it is reported to be offline
2018-01-30 15:14:11 #6(rspamd_proxy) <cfd4cb>; lua; greylist.lua:348: greylisted until "Tue, 30 Jan 2018 15:19:11 GMT", new record
2018-01-30 15:14:11 #6(rspamd_proxy) <cfd4cb>; proxy; lua_task_set_pre_result: <72e3cbe057d72971f453d93d31206447@subdomain.example.com>: set pre-result to soft reject: 'Try again later'
2018-01-30 15:14:11 #6(rspamd_proxy) <cfd4cb>; proxy; rspamd_task_write_log: id: <72e3cbe057d72971f453d93d31206447@subdomain.example.com>, qid: <0E14041643>, ip: 1.1.1.1, from: <client@example.com>, (default: F (soft reject): [13.02/15.00] [ONCE_RECEIVED_STRICT(4.00){},R_UNDISC_RCPT(3.00){},BAYES_HAM(-2.78){99.04%;},HFILTER_HOSTNAME_UNKNOWN(2.50){},DMARC_POLICY_REJECT(2.00){example.com : No valid SPF, No valid DKIM;reject;},FORGED_RECIPIENTS(2.00){},RDNS_NONE(1.00){},R_SPF_FAIL(1.00){-all;},MIME_HTML_ONLY(0.20){},ONCE_RECEIVED(0.10){},ARC_NA(0.00){},ASN(0.00){asn:197019, ipnet:1.1.1.0/20, country:--;},DIRECT_TO_MX(0.00){PHPMailer 5.2.25 (https://github.com/PHPMailer/PHPMailer);},FROM_EQ_ENVFROM(0.00){},FROM_HAS_DN(0.00){},GREYLIST(0.00){greylisted;Tue, 30 Jan 2018 15:19:11 GMT;new record;},HAS_PHPMAILER_SIG(0.00){},HAS_REPLYTO(0.00){client@domain.com;},IP_SCORE(0.00){asn: 197019(0.13), country: --(0.04);},RCPT_COUNT_ONE(0.00){1;},RCVD_COUNT_ZERO(0.00){0;},RCVD_TLS_ALL(0.00){},REPLYTO_DOM_NEQ_FROM_DOM(0.00){},R_DKIM_NA(0.00){},TO_DN_ALL(0.00){}]), len: 1157, time: 888.000ms real, 4.762ms virtual, dns req: 28, digest: <2b01e01cfe056645f51157bda188d6f3>, rcpts: <out@example.com>
2018-01-30 15:14:11 #6(rspamd_proxy) <cfd4cb>; proxy; rspamd_protocol_http_reply: regexp statistics: 60 pcre regexps scanned, 6 regexps matched, 163 regexps total, 7 regexps cached, 5.05k bytes scanned using pcre, 5.05k bytes scanned total
2018-01-30 15:14:12 #6(rspamd_proxy) <082243>; proxy; proxy_milter_finish_handler: finished milter connection
```

**Comment by kaiyou** [comment]
Oh, rmilter indeed still does Greylisting even for whitelisted addresses. I think we should add relay domains to the whitelist ; if you succeed in configuring it properly, please pull request. I am marking this as a bug in the meanwhile :)

**Comment by kaiyou** [comment]
This should be fixed with #612 being merged.
